lib/beacon/content.ex

```diff
@@ -1667,6 +1680,108 @@ defmodule Beacon.Content do
     end
   end
 
+  # ERROR PAGES
+
+  @doc """
+  Returns an `%Ecto.Changeset{}` for tracking error page changes.
+
+  ## Example
+
+      iex> change_error_page(error_page, %{status: 404})
+      %Ecto.Changeset{data: %ErrorPage{}}
+
+  """
+  @doc type: :error_pages
+  @spec change_error_page(ErrorPage.t(), map()) :: Changeset.t()
+  def change_error_page(%ErrorPage{} = error_page, attrs \\ %{}) do
+    ErrorPage.changeset(error_page, attrs)
+  end
+
+  @doc """
+  Returns the error page for a given site and status code, or `nil` if no matching error page exists.
+  """
+  @doc type: :error_pages
+  @spec get_error_page(Site.t(), integer()) :: ErrorPage.t() | nil
+  def get_error_page(site, status) do
+    Repo.one(
+      from e in ErrorPage,
+        where: e.site == ^site,
+        where: e.status == ^status
+    )
+  end
+
+  @doc """
+  Lists all error pages for a given site.
+  """
+  @doc type: :error_pages
+  @spec list_error_pages(Site.t()) :: [ErrorPage.t()]
+  def list_error_pages(site) do
+    Repo.all(
+      from e in ErrorPage,
+        where: e.site == ^site,
+        order_by: e.status
+    )
+  end
+
+  @doc """
+  Creates a new error page.
+  """
+  @doc type: :error_pages
+  @spec create_error_page(%{site: Site.t(), status: integer(), template: binary(), layout_id: Ecto.UUID.t()}) ::
```

Comment:
can use that type here too

```suggestion
  @spec create_error_page(%{site: Site.t(), status: status_code(), template: binary(), layout_id: Ecto.UUID.t()}) ::
```